content/sensu-go/5.17/reference/secrets.md

```diff
@@ -0,0 +1,220 @@
+required     | Required for secrets configuration in `wrapped-json` or `yaml` format.
+type         | Map of key-value pairs
+example      | {{< highlight shell >}}
+"spec": {
+  "id": "ANSIBLE_TOKEN",
+  "provider": "ansible_vault"
+}
+{{< /highlight >}}
+
+### Metadata attributes
+
+name         |      |
+-------------|------
+description  | The name for the secret that is used internally by Sensu.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"name": "sensu-ansible-token"{{< /highlight >}}
+
+namespace    |      |
+-------------|------
+description  | [Sensu RBAC namespace][9] that the secret belongs to.
+--
+
+
+.
+required     | true
+type         | String
+example      | {{< highlight shell >}}"namespace": "default"{{< /highlight >}}
+
+### Spec attributes
+
+id           | 
+-------------|------ 
+description  | The identifying key for the provider to retrieve the secret.
+required     | true
+type         | String
+default      | `false`
+example      | {{< highlight shell >}}"id": "secret/ansible#token"{{< /highlight >}}
+
+provider     | 
+-------------|------ 
+description  | The name of the provider with the secret.
+required     | true
+type         | String
+default      | `false`
+example      | {{< highlight shell >}}"provider": "ansible_vault"{{< /highlight >}}
+
+## Secret configuration
+
+Use the [Secrets API][2] and [sensuctl][3] to create, view, and manage your secrets configuration.
+To manage secrets configuration with sensuctl, configure sensuctl as the default [`admin` user][6].
+
+The [standard sensuctl subcommands][4] are available for secrets (list, info, and delete).
+
+To list all secrets:
+
+{{< highlight shell >}}
+sensuctl secret list
+{{< /highlight >}}
+
+To see a secret's status:
+
+{{< highlight shell >}}
+sensuctl secret info SECRET_NAME
+{{< /highlight >}}
+
+To delete a secret:
+
+{{< highlight shell >}}
+sensuctl secret delete SECRET_NAME
+{{< /highlight >}}
+
+## Secret examples
+
+A secret resource definition refers to a secrets `id` and a secrets `provider`.
+Read the [secrets provider reference][7] for the provider specification.
+
+{{< language-toggle >}}
+
+{{< highlight yml >}}
+---
+type: Secret
+api_version: secrets/v1
+metadata:
+  name: sensu-ansible-token
+  namespace: default
+spec:
+  id: ANSIBLE_TOKEN
+  provider: env
+{{< /highlight >}}
+
+{{< highlight json >}}
+{
+  "type": "Secret",
+  "api_version": "secrets/v1",
+  "metadata": {
+    "name": "sensu-ansible-token",
+    "namespace": "default"
+  },
+  "spec": {
+    "id": "ANSIBLE_TOKEN",
+    "provider": "env"
+  }
+}
+{{< /highlight >}}
+
+{{< /language-toggle >}}
+
+Secrets that target a HashiCorp Vault always have an `id` as shown in the following example:
+
+{{< language-toggle >}}
+
+{{< highlight yml >}}
+---
+type: Secret
+api_version: secrets/v1
+metadata:
+  name: sensu-ansible-token
+  namespace: default
+spec:
+  id: secret/website#database
```

Comment:
YML thinks this is a comment. Does surrounding the line with single quotes resolve this?
```suggestion
  id: 'secret/website#database'
```

<img width="319" alt="Screen Shot 2020-01-22 at 10 01 00 AM" src="https://user-images.githubusercontent.com/10634078/72920439-2054a780-3cfe-11ea-8712-347709873672.png">

Reply:
Yes, that fixes it:
![yml-update](https://user-images.githubusercontent.com/19190208/72925076-15127380-3d20-11ea-814a-a19b06c9c400.png)